### PR TITLE
Connect buy flow confirmation flow

### PR DESCRIPTION
**Pull request by kenan-mamedoff**

### CODE REVIEW

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/BuyCoordinator.swift [hunk 1]
```diff
@@ -28,19 +33,71 @@ class BuyCoordinator: BaseCoordinator, BuyRoutes, BillingAddressRoutes, OrderPre
...
+    func showThreeDSecure(url: URL, flow: Flow) {
         let webViewController = SimpleWebViewController(url: url)
         webViewController.setup(with: .init(title: "3D Secure")) // TODO: Localize
         webViewController.didDismiss = { [weak self] in
-            let vc = self?.navigationController.children.first(where: { $0 is BillingAddressViewController }) as? BillingAddressViewController
-            vc?.interactor?.checkThreeDSecureStatus(viewAction: .init())
+            switch flow {
+            case .addCard:
+                let vc = self?.navigationController.children.first(where: { $0 is BillingAddressViewController }) as? BillingAddressViewController
+                vc?.interactor?.checkThreeDSecureStatus(viewAction: .init())
```
Comment (rcresnik) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/BuyCoordinator.swift:43]: if you would use the existing fetch cycle, you would not need this magic here (considering the VC's dont need any vars but only need 1 interactor method to be called..)

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/BuyCoordinator.swift [hunk 2]
```diff
@@ -28,19 +33,71 @@ class BuyCoordinator: BaseCoordinator, BuyRoutes, BillingAddressRoutes, OrderPre
...
+            vc.secondCallback = { [weak self] in
+                self?.showSwapDetails(exchangeId: paymentReference)
+            }
+        }
+    }
+    
+    func showFailure() {
+        open(scene: Scenes.Failure) { vc in
+            vc.firstCallback = { [weak self] in
+                // TODO: How does this work? How do we navigate to cards screen?
```
Comment (rcresnik) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/BuyCoordinator.swift:75]: pop2Root + present card selection (without animation)

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewInteractor.swift [hunk 3]
```diff
@@ -51,19 +51,50 @@ class OrderPreviewInteractor: NSObject, Interactor, OrderPreviewViewActions {
...
+                
+                self?.handlePresentSubmit()
+                
+            case .failure(let error):
+                self?.presenter?.presentError(actionResponse: .init(error: error))
+            }
+        }
+    }
+    // TODO: Merge the logic with Billing Details. They are the same things. The whole card and 3D secure flow should be reusable.
+    private func handlePresentSubmit() {
```
Comment (rcresnik) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewInteractor.swift:90]: maybe its time for u to make ur first reusable cycle :)

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift [hunk 4]
```diff
@@ -57,7 +57,7 @@ final class OrderPreviewPresenter: NSObject, Presenter, OrderPreviewActionRespon
             .termsAndConditions: [
                 LabelViewModel.text("By placing this order you agree to our Terms and Conditions")
             ],
-            .confirm: [
+            .submit: [
```
Comment (rcresnik) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift:60]: just curious.. why the rename? 🤔

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift [hunk 5]
```diff
@@ -71,19 +71,19 @@ final class OrderPreviewPresenter: NSObject, Presenter, OrderPreviewActionRespon
...
-            presentError(actionResponse: .init(error: GeneralError(errorMessage: "No redirect url!")))
-            return
-        }
-        viewController?.displayConfirm(responseDisplay: .init(url: url))
+    func presentThreeDSecure(actionResponse: OrderPreviewModels.ThreeDSecure.ActionResponse) {
+        viewController?.displayThreeDSecure(responseDisplay: .init(url: actionResponse.url))
     }
     
-    func presentCvv(actionResponse: OrderPreviewModels.CvvValidation.ActionResponse) {
+    func presentCvv(actionResponse: OrderPreviewModels.CVVValidation.ActionResponse) {
```
Comment (rcresnik) [inline brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift:78]: that is not how we "uppercase' abbreviations

**Review by rcresnik — APPROVED** [review]

File: brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift (hunk 4 above)
Comment (kenan-mamedoff) [reply brd-ios/breadwallet/src/Redesign/Scenes/Buy/OrderPreview/OrderPreviewPresenter.swift:60]: just to keep it unified with the billing details screen. will make it reusable.

**Pull request merged**